### ISSUE TITLE
La Vitrine Sources - add aggregator that is filtered by taxonomy criteria for Signe Laval

### ISSUE THREAD

**Issue by MaudeFJ** [culturecreates/tech-support-tickets#94]
### User story
Signe Laval would like to import (using an aggregator) certain events from the La Vitrine source in Artsdata. The events in question would be a filtered subset of the La Vitrine events, the ones that feature Lavallois artists. In other words the La Vitrine aggregator from Artsdata would only include events with Lavallois artists performing.

Your 'La Vitrine' aggregator would only include events that meet this criteria:
-performed by "lavallois artist"

The Signe Laval CMS would be the source of the latest list of Lavallois artists (people and organizations) via these 2 taxonomies used in people/orgs:
Orgs: the dynamic taxonomy is called "Ville de l'organisation:
People: the dynamic taxonomy is called "Ville de résidence"
NOTE: the names of the taxonomies are subject to change.
These taxonomies are used to identify which people/Organizations in the Signe Laval CMS are Lavallois. So this information would have to be available to identify which events to import from Artsdata.

_Could we automate entry of lavallois artist event's that are in arts data in my cms ?
Like if Adam Jonhson as an event at la place des arts it can enter straight up in brouillon in my cms !_

**Comment by troughc** [comment]
@MaudeFJ does this capture what you want?
Signe Laval would like to import (using an aggregator) certain events from the La Vitrine source in Artsdata. The events in question would be a filtered subset of the La Vitrine events, the ones that feature Lavallois artists. In other words the La Vitrine aggregator from Artsdata would only include events with Lavallois artists performing.

Your 'La Vitrine' aggregator would only include events that meet this criteria:
- performed by "lavallois artist"

Signe Laval would provide the list of Lavallois artists (people and organizations).

**Comment by MaudeFJ** [comment]
Yes ! 
But Ideally we would not have to provide a list of lavallois artist, because this info would be embed with the people and organization already in arts data since i'm already putting it in the cms !  
<img width="820" alt="Capture d’écran, le 2024-10-03 à 15 47 39" src="https://github.com/user-attachments/assets/bdf10ab4-e681-4076-b819-453069fba5d9">

**Comment by troughc** [comment]
> But Ideally we would not have to provide a list of lavallois artist, because this info would be embed with the people and organization already in arts data since i'm already putting it in the cms !
Yes, that is what I meant. The Signe Laval data includes the taxonomy that identifies a lavallois.

Thanks for the user story.

**Comment by troughc** [comment]
@MaudeFJ please take a look at the first comment again to see if it makes sense. if yes, just give me a thumbs up. :)

**Comment by troughc** [comment]
@sahalali let's discuss this and what it would entail in terms of development.

**Comment by MaudeFJ** [comment]
Yes well explain 
About the taxonomie 
Orgs: Ville de l'organisation
People: Lieu de travail 
I will
 change lieu de travail for Ville de résidence because some artists were confuse

**Comment by suhail-aliyar** [comment]
If I understand correctly, the aggregator should be able to configure a source (a workflow) to import only events "performed by" a set of people with matching property 'Lieu de travail' value or organization with matching property 'Ville de l'organisation' value in the CMS.

Design:
Step 1: Fetch all people and organizations that match a given condition in the configuration file (now called the mapping file). The aggregator should be able to fetch a list of people and organizations from the CMS. 
Step 2: Iterate events fetched from CMS as of today and skip those events that do not match the condition (in this case "performed by") in the configuration file.

This will take 3 days of effort to implement and test.

**Comment by suhail-aliyar** [comment]
@troughc Please check the estimate and prioritize.

**Comment by dev-aravind** [comment]
@troughc 

> Signe Laval would like to import (using an aggregator) certain events from the La Vitrine source in Artsdata

Can you share the URL to the La Vitrine source, I was not able to find it in artsdata.

**Comment by saumier** [comment]
@dev-aravind @troughc The LaVitrine source is not available. Once CAPACOA gives us the green light the Artsdata team will add LaVitrine to Artsdata. Probably after Christmas.

**Comment by troughc** [comment]
To do: find a source of identified events that include lavallois artists (organizations and/or people in the performer field). Maybe Tout Culture, salle odyssee, minotaure, ville de gatineau, **place des arts**
Test using this source's events:
1.  Pick the source site with identified performer that is already in Artsdata.
2. Create a new aggregate source for Signe laval using the new filter based on the custom taxonomies for people and organizations (people and organizations that reside in Laval)
3. Run aggregator to see if it works

In Signe Laval, here are examples of lavallois artists
Example person with correct selection in taxonomy Ville de résidence: https://cms.footlight.io/dashboard/63457978637093005071a053/people/670837b89ab04f006428d02e
Example organization with correct selection in taxonomy Ville de l'organisation:
https://cms.footlight.io/dashboard/63457978637093005071a053/organizations/6442813ef45aca0064127876

**Comment by troughc** [comment]
@MaudeFJ we need some real examples of your artists performing in events at other places that are sources in Artsdata. And they ahave to be identified. Sources that could work in our tests are:
Tout Culture, 
salle odyssee, 
minotaure, 
ville de gatineau,
Place des arts

TO DO: Please provide some example events at one or two of these places that have your artist performing (a few events)

**Comment by MaudeFJ** [comment]
Hello, j'en ai trouvé juste 1 à date :
https://placedesarts.com/fr/evenement/half-moon-run-osm-pop

J'aurais besoin d'aide... 
Genre il y a des lieux/ organisations qui sont dans arts data où des artiste lavallois performent, mais je n'arrive pas à voir les événement qui sont link avec ces lieux/organisations dans arts data... Est-ce qu'il y a moyen de faire ça ? 
( je pense surtout à ces spectacles : https://bluff.qc.ca/calendrier/ genre à la maison théâtre ou au théâtre denise pelletier)

**Comment by MaudeFJ** [comment]
https://kg.artsdata.ca/entity?uri=http%3A%2F%2Fkg.artsdata.ca%2Fresource%2FK23-1785

**Comment by troughc** [comment]
@saumier please see Maude's question above.

**Comment by saumier** [comment]
@troughc I think this issue is stale. Please close if you agree.
